pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -0,0 +1,837 @@
+	BuildLabel                   = "kpack.io/build"
+	DOCKERSecretAnnotationPrefix = "kpack.io/docker"
+	GITSecretAnnotationPrefix    = "kpack.io/git"
+
+	cacheDirName              = "cache-dir"
+	layersDirName             = "layers-dir"
+	platformDir               = "platform-dir"
+	homeDir                   = "home-dir"
+	workspaceDir              = "workspace-dir"
+	imagePullSecretsDirName   = "image-pull-secrets-dir"
+	builderPullSecretsDirName = "builder-pull-secrets-dir"
+
+	notaryDirName = "notary-dir"
+	reportDirName = "report-dir"
+
+	networkWaitLauncherDir = "network-wait-launcher-dir"
+
+	buildChangesEnvVar = "BUILD_CHANGES"
+	platformAPIEnvVar  = "CNB_PLATFORM_API"
+)
+
+type BuildPodImages struct {
+	BuildInitImage         string
+	CompletionImage        string
+	RebaseImage            string
+	BuildInitWindowsImage  string
+	CompletionWindowsImage string
+}
+
+func (bpi *BuildPodImages) buildInit(os string) string {
+	switch os {
+	case "windows":
+		return bpi.BuildInitWindowsImage
+	default:
+		return bpi.BuildInitImage
+	}
+}
+
+func (bpi *BuildPodImages) completion(os string) string {
+	switch os {
+	case "windows":
+		return bpi.CompletionWindowsImage
+	default:
+		return bpi.CompletionImage
+	}
+}
+
+type BuildPodBuilderConfig struct {
+	StackID      string
+	RunImage     string
+	Uid          int64
+	Gid          int64
+	PlatformAPIs []string
+	OS           string
+}
+
+var (
+	sourceVolume = corev1.VolumeMount{
+		Name:      workspaceDir,
+		MountPath: "/workspace",
+	}
+	homeVolume = corev1.VolumeMount{
+		Name:      homeDir,
+		MountPath: "/builder/home",
+	}
+	platformVolume = corev1.VolumeMount{
+		Name:      platformDir,
+		MountPath: "/platform",
+	}
+	cacheVolume = corev1.VolumeMount{
+		Name:      cacheDirName,
+		MountPath: "/cache",
+	}
+	layersVolume = corev1.VolumeMount{
+		Name:      layersDirName,
+		MountPath: "/layers",
+	}
+	projectMetadataVolume = corev1.VolumeMount{
+		Name:      layersDirName,
+		MountPath: "/projectMetadata",
+	}
+	homeEnv = corev1.EnvVar{
+		Name:  "HOME",
+		Value: "/builder/home",
+	}
+	imagePullSecretsVolume = corev1.VolumeMount{
+		Name:      imagePullSecretsDirName,
+		MountPath: "/imagePullSecrets",
+		ReadOnly:  true,
+	}
+	builderPullSecretsVolume = corev1.VolumeMount{
+		Name:      builderPullSecretsDirName,
+		MountPath: "/builderPullSecrets",
+		ReadOnly:  true,
+	}
+	notaryV1Volume = corev1.VolumeMount{
+		Name:      notaryDirName,
+		MountPath: "/var/notary/v1",
+		ReadOnly:  true,
+	}
+	reportVolume = corev1.VolumeMount{
+		Name:      reportDirName,
+		MountPath: "/var/report",
+		ReadOnly:  false,
+	}
+	networkWaitLauncherVolume = corev1.VolumeMount{
+		Name:      networkWaitLauncherDir,
+		MountPath: "/networkWait",
+		ReadOnly:  false,
+	}
+)
+
+type stepModifier func(corev1.Container) corev1.Container
+
+func (b *Build) BuildPod(images BuildPodImages, secrets []corev1.Secret, taints []corev1.Taint, config BuildPodBuilderConfig) (*corev1.Pod, error) {
```

Comment:
Should me delete the equivalent of these domain methods in v1alpha1?

Reply:
You are right I guess. I have removed them, but in a separate commit. So it is easier to revert again. I can squash it, once this is ready to merge.

Reply:
Actually we'd like to keep them to minimize the manual effort when doing a version bump. Also we should not touch old versions to not break existing code (maybe deprecate the methods instead).